android/app/src/main/java/org/mozilla/magnet/api/Api.java

```diff
@@ -0,0 +1,148 @@
+package org.mozilla.magnet.api;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.telecom.Call;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.StringRequest;
+import com.android.volley.toolbox.Volley;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by wilsonpage on 28/10/2016.
+ */
+
+public abstract class Api {
+    private HashMap<String,Api> routes = new HashMap<>();
+    private RequestQueue mQueue;
+    private Context mContext;
+
+    public Api(Context context) {
+        mContext = context;
+    }
+
+    protected void mount(String namespace, Api api) {
+        routes.put(namespace, api);
+    }
+
+    private Api find(String path) {
+        for (Map.Entry<String, Api> entry : routes.entrySet()) {
```

Comment:
Why do we have this loop? Can we just check if the key exists and access to it directly right?